flask/datahandle.py

```diff
@@ -82,7 +125,71 @@ def datahandle(date_in,loc_in, meal_in):
     
     #checking if specified meal available
     if checkMealAvailable(data, meal_in):
+        print(getCoursesAndItems(data))
         return getCoursesAndItems(data)
     else:
         return "No meal is available."
 
+#Handle meal item data request
+def requestItem(date_in,loc_in, meal_in, item_in):
+    url = 'http://api.studentlife.umich.edu/menu/xml2print.php?controller=&view=json'
+    location = '&location='
+    date = '&date='
+    meal = '&meal='
+
+    #API url concatenation
+    location += loc_in
+    date += str(date_in)
+    url = url + location + date + meal
+    url = removeSpaces(url)
+
+    if meal_in == '':
+        mealEntered = False
+    else:
+        mealEntered = True
+    
+    #fetching json
+    data = requests.get(url).json()
+    
+    possiblematches = []
+    
+    firstRound = True
+    
+    #Loop through meals
+    for i in range(len(data['menu']['meal'])):
```

Comment:
Can you just use something like

`for meal  in data['menu']['meal'] `

here without the numeric index? I see everywhere just referring to `data['menu']['meal'][i]`. This would make it more pythonic and cleaner.

Reply:
Oops, it used to be necessary the way I was doing it in a previous version. Overlooked that, thanks.